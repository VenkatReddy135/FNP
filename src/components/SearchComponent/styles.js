import { makeStyles } from "@material-ui/core/styles";

/**
 * makeStyles hook of material-ui to apply style for Login form
 *
 *  @function useStyles
 */
const useStyles = makeStyles(() => ({
  categoryUrl: {
    width: "255px",
    marginTop: "8px",
  },
  catBasic: {
    width: "240px",
    marginRight: "0px",
  },
  pageSimpleForm: {
    display: "flex",
    alignItems: "center",
  },
  searchBtn: {
    margin: "0px 0px 25px 50px",
  },
}));

export default useStyles;
