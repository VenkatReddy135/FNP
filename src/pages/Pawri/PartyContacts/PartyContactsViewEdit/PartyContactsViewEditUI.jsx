/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useMemo } from "react";
import PropTypes from "prop-types";
import {
  useTranslate,
  SimpleForm,
  SimpleShowLayout,
  TextInput,
  NumberInput,
  BooleanInput,
  AutocompleteInput,
  required,
} from "react-admin";
import { useParams } from "react-router-dom";
import { Grid, Typography, Divider, IconButton, TextField } from "@material-ui/core";
import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
import useStyles from "../../../../assets/theme/common";
import {
  maxLength,
  validateLetter,
  handleInvalidCharsInNumberInput,
  phoneNoValidation,
  validateEmail,
} from "../../../../utils/validationFunction";
import formatDateValue, { fetchDateString } from "../../../../utils/formatDateTime";
import { EMAIL_ADDRESS, PHONE_NUMBER, POSTAL_ADDRESS, IS_EMPTY, COUNTRY_INDIA } from "../PartyContactsConstants";
import { renderContactInfoFields } from "../HelperFunctions";
import CustomToolbar from "../../../../components/CustomToolbar";
import DateTimeInput from "../../../../components/CustomDateTimeV2";
import Breadcrumbs from "../../../../components/Breadcrumbs";

/**
 * Component to render the View/Edit Page of Party Contacts
 *
 * @param {object} props needed for view and edit
 * @returns {React.ReactElement} all the props required by the Party Contacts - View/Edit
 */
const PartyContactsViewEditUI = (props) => {
  const translate = useTranslate();
  const { partyId } = useParams();
  const classes = useStyles();
  const {
    responseData,
    isPincodeInvalid,
    isEditable,
    switchToEditHandler,
    contactPurposes,
    cancelHandler,
    updateContact,
    handleChange,
    countryList,
    stateList,
    citiesList,
    countryCodes,
    handleCityStateCountryChange,
    handleDate,
    handlePhoneChange,
  } = props;
  const contactTypeEmail = responseData && responseData?.contactType === EMAIL_ADDRESS;
  const contactTypePhone = responseData && responseData?.contactType === PHONE_NUMBER;
  const contactTypePostal = responseData && responseData?.contactType === POSTAL_ADDRESS;
  const { fromDate, createdBy, createdAt, updatedBy, updatedAt, primary, status } = responseData;
  const gridTitle = responseData?.contactType;
  const invalidPincodeMessage = isPincodeInvalid ? translate("pincode_validation") : IS_EMPTY;
  const pincodeValidationMessageText = !responseData.pincode ? translate("pincode_required") : invalidPincodeMessage;

  const breadcrumbs = [
    {
      displayName: translate("party_management"),
      navigateTo: `/parties/search`,
    },
    {
      displayName: `${partyId}`,
      navigateTo: `/${window.REACT_APP_PARTY_SERVICE}/parties/search/${partyId}/show/contact-info`,
    },
    { displayName: `${gridTitle}` },
  ];
  const emailValidation = [
    required(translate("email_validation_required")),
    validateEmail(translate("invalid_email_validation_message")),
  ];
  const validateToName = [
    required(translate("nameValidation")),
    maxLength(50, translate("nameValidationMax")),
    validateLetter(translate("ToName_validation_message")),
  ];
  const attentionNameValidation = [
    maxLength(50, translate("nameValidationMax")),
    validateLetter(translate("attention_validation_message")),
  ];
  const phoneValidation = [
    phoneNoValidation(responseData.countryCodeId, translate("phone_number_error")),
    required(translate("phone_validation_required")),
  ];
  /**
   *@function toDateValidation to validate the toDate field
   * @param {Date} value date value to be validated
   * @returns {(string | undefined)} returns string or undefined
   */
  const toDateValidation = (value) => {
    return value && fetchDateString(new Date(value)) <= fetchDateString(new Date(responseData.fromDate))
      ? translate("to_date_validation_message")
      : undefined;
  };
  /**
   * @returns {React.Component} return component
   */
  const RenderTitle = useMemo(
    () => (
      <>
        <Breadcrumbs breadcrumbs={breadcrumbs} />
        <Grid container direction="row" className={classes.titleGridStyle} justify="space-between">
          <Typography variant="h5" className={classes.gridStyle}>
            {gridTitle}
          </Typography>
        </Grid>
        <Divider variant="fullWidth" className={classes.dividerStyle} />
      </>
    ),
    [gridTitle],
  );
  /**
   * @returns {React.Component} return component
   */
  const RenderFromAndThroughDate = useMemo(
    () => (
      <>
        <Grid direction="row" container md={12} spacing={10} className={classes.plusButton}>
          <Grid item className={classes.gridMarginStyle}>
            <DateTimeInput source="fromDate" label={translate("from_date")} className={classes.dateField} disabled />
          </Grid>
          <Grid item className={classes.textFieldItem}>
            <DateTimeInput
              source="throughDate"
              label={translate("through_date")}
              className={classes.customDateTimeInput}
              onChange={handleDate}
              validate={toDateValidation}
              disabled={!isEditable}
            />
          </Grid>
        </Grid>
      </>
    ),
    [fromDate, isEditable],
  );
  /**
   * @returns {React.Component} return component
   */
  const RenderPostalAddress = useMemo(
    () => (
      <>
        <Grid item container direction="row" spacing={7}>
          <Grid item>
            <TextInput
              source="toName"
              label={translate("to_name")}
              className={classes.textInputField}
              disabled={!isEditable}
              validate={validateToName}
            />
          </Grid>
          <Grid item className={classes.customMargin}>
            <TextInput
              source="attentionName"
              label={translate("attention_name")}
              className={classes.textInputField}
              disabled={!isEditable}
              validate={attentionNameValidation}
            />
          </Grid>
          <Grid item>
            <NumberInput
              source="doorbell"
              label={translate("doorbell")}
              className={classes.numberInputField}
              disabled={!isEditable}
              onKeyDown={handleInvalidCharsInNumberInput}
            />
          </Grid>
        </Grid>
        <Grid item className={classes.customMargin}>
          <TextInput
            source="addressLine1"
            label={translate("address_line_1")}
            className={classes.addressInputField}
            disabled={!isEditable}
            validate={required(translate("address1_validation_required"))}
          />
        </Grid>
        <Grid item className={classes.customMargin}>
          <TextInput
            source="addressLine2"
            label={translate("address_line_2")}
            className={classes.addressInputField}
            disabled={!isEditable}
          />
        </Grid>
        <Grid item container direction="row" spacing={7} className={classes.customMargin}>
          <Grid item>
            <AutocompleteInput
              source="countryId"
              choices={countryList}
              className={classes.textInputField}
              onChange={(e) => handleCityStateCountryChange("countryId", e)}
              validate={required(translate("country_required"))}
              disabled={!isEditable}
            />
          </Grid>
          {stateList.length === 0 ? (
            <Grid item md={2} className={classes.disabledText}>
              <Typography variant="caption">{translate("state")}</Typography>
              <Typography variant="subtitle2">{translate("not_available")}</Typography>
            </Grid>
          ) : (
            <Grid item>
              <AutocompleteInput
                source="stateId"
                choices={stateList}
                className={classes.textInputField}
                disabled={!isEditable}
                onChange={(e) => handleCityStateCountryChange("stateId", e)}
                validate={stateList.length !== 0 && required(translate("state_required"))}
              />
            </Grid>
          )}
          <Grid item md={2}>
            {citiesList?.length > 0 ? (
              <AutocompleteInput
                source="cityId"
                choices={citiesList}
                validate={required(translate("city_required"))}
                className={classes.textInputField}
                onChange={(e) => handleCityStateCountryChange("cityId", e)}
                disabled={!isEditable || !responseData.countryId}
                shouldRenderSuggestions={(val) => {
                  return val && val.trim().length > 1;
                }}
              />
            ) : (
              <TextInput
                source="cityId"
                label={translate("city")}
                className={[classes.autoCompleteStyle, classes.disabledText]}
                validate={required(translate("city_required"))}
                onChange={(e) => handleCityStateCountryChange("cityId", e.target.value)}
                disabled={!isEditable || !responseData.countryId}
              />
            )}
          </Grid>
        </Grid>
        {isEditable ? (
          <Grid item className={classes.customMargin}>
            <TextField
              name="pincode"
              label={translate("pincode")}
              className={classes.autoCompleteStyle}
              onChange={handleChange}
              value={responseData.pincode}
              required={responseData.countryId === COUNTRY_INDIA}
              helperText={responseData.countryId === COUNTRY_INDIA && pincodeValidationMessageText}
              error={responseData.countryId === COUNTRY_INDIA && (isPincodeInvalid || !responseData.pincode)}
            />
          </Grid>
        ) : (
          <Grid item className={classes.gridMarginStyle}>
            <Typography variant="caption">{translate("pincode")}</Typography>
            <Typography variant="subtitle2">{responseData.pincode}</Typography>
          </Grid>
        )}
      </>
    ),
    [citiesList, countryList, isEditable, responseData.pincode, stateList, pincodeValidationMessageText],
  );
  return (
    <>
      {RenderTitle}
      <SimpleShowLayout component="div">
        <SimpleForm
          initialValues={responseData}
          save={updateContact}
          toolbar={
            isEditable ? <CustomToolbar onClickCancel={cancelHandler} saveButtonLabel={translate("update")} /> : null
          }
        >
          <Grid item container direction="row" spacing={7} mb={1}>
            <Grid item>
              <TextInput
                source="contactType"
                label={translate("contact_type")}
                disabled
                className={classes.textInputField}
              />
            </Grid>
            <Grid item>
              <AutocompleteInput
                source="contactPurposeId"
                choices={contactPurposes}
                label={translate("contact_purpose")}
                value=""
                className={classes.autoCompleteItem}
                disabled={!isEditable}
              />
            </Grid>
            {contactTypeEmail && (
              <Grid item>
                <TextInput
                  source="email"
                  label={translate("contact_email_address")}
                  type="email"
                  className={classes.textInputField}
                  disabled={!isEditable}
                  validate={emailValidation}
                />
              </Grid>
            )}
            {contactTypePhone && (
              <>
                <Grid item>
                  <AutocompleteInput
                    label={translate("code")}
                    source="countryCodeId"
                    choices={countryCodes}
                    disabled={!isEditable}
                    className={classes.autoCompleteCode}
                    onChange={(e) => handlePhoneChange("countryCodeId", e)}
                  />
                </Grid>
                <Grid item>
                  <NumberInput
                    source="phone"
                    label={translate("contact_phone_number")}
                    className={classes.numberInputField}
                    disabled={!isEditable}
                    validate={phoneValidation}
                    onKeyDown={handleInvalidCharsInNumberInput}
                    onChange={(e) => handlePhoneChange("phone", e.target.value)}
                  />
                </Grid>
              </>
            )}
            {!isEditable && (
              <Grid item className={classes.label}>
                <Typography variant="caption">{translate("status")}</Typography>
                <Typography variant="subtitle2">{status ? translate("active") : translate("expired")}</Typography>
              </Grid>
            )}
            {isEditable ? null : (
              <Grid item className={classes.buttonAlignment}>
                <IconButton onClick={switchToEditHandler}>
                  <EditOutlinedIcon />
                </IconButton>
              </Grid>
            )}
          </Grid>
          {!contactTypePostal &&
            (isEditable ? (
              <Grid item className={classes.gridMarginStyle}>
                <BooleanInput label={translate("primary")} source="primary" />
              </Grid>
            ) : (
              <Grid item className={classes.gridMarginStyle}>
                <Typography variant="caption">{translate("primary")}</Typography>
                <Typography variant="subtitle2">{primary?.toString()}</Typography>
              </Grid>
            ))}
          {contactTypePostal && RenderPostalAddress}
          {RenderFromAndThroughDate}
          {!isEditable &&
            renderContactInfoFields(
              classes,
              translate("created_by"),
              createdBy,
              translate("created_on"),
              formatDateValue(createdAt),
            )}
          {!isEditable &&
            renderContactInfoFields(
              classes,
              translate("modified_by"),
              updatedBy,
              translate("modified_date"),
              formatDateValue(updatedAt),
            )}
        </SimpleForm>
      </SimpleShowLayout>
    </>
  );
};
PartyContactsViewEditUI.propTypes = {
  responseData: PropTypes.objectOf(PropTypes.any).isRequired,
  isEditable: PropTypes.bool.isRequired,
  isPincodeInvalid: PropTypes.bool.isRequired,
  switchToEditHandler: PropTypes.func.isRequired,
  contactPurposes: PropTypes.arrayOf(PropTypes.any).isRequired,
  countryList: PropTypes.arrayOf(PropTypes.any).isRequired,
  stateList: PropTypes.arrayOf(PropTypes.any).isRequired,
  citiesList: PropTypes.arrayOf(PropTypes.any).isRequired,
  cancelHandler: PropTypes.func.isRequired,
  updateContact: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleCityStateCountryChange: PropTypes.func.isRequired,
  countryCodes: PropTypes.arrayOf(PropTypes.any).isRequired,
  handleDate: PropTypes.func.isRequired,
  handlePhoneChange: PropTypes.func.isRequired,
};
export default React.memo(PartyContactsViewEditUI);
