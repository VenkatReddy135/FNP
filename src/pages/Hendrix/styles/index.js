import { makeStyles } from "@material-ui/core/styles";

/**
 * makeStyles hook of material-ui to apply common style for any components
 *
 *  @function useStyles
 */
const useStyles = makeStyles({
  pageSimpleForm: {
    display: "flex",
    alignItems: "center",
  },
  dropdownContainer: {
    width: "180px",
  },
  saveBtn: {
    margin: "0px 0px 25px 50px",
  },
  updateBtn: {
    position: "absolute",
    top: "90px",
    right: "30px",
  },
  container: {
    padding: "20px 10px 40px",
    display: "flex",
    alignItems: "center",
  },
  textField: {
    width: "400px",
  },
  textFields: {
    marginRight: "25px",
  },
  listBtn: {
    margin: "0px 0px 25px 50px",
  },
  templateDropdown: {
    width: "100%",
    minWidth: "350px",
  },
  select: {
    "& ul": {
      width: "330px",
      maxHeight: "150px",
    },
  },
  ruleContainer: {
    margin: "150px 0px 0px 100px",
  },
  headerActionButton: {
    position: "absolute",
    right: "20px",
  },
  actionBtn: {
    position: "relative",
    top: "90px",
  },
  actionBtndetail: {
    position: "relative",
    top: "90px",
    marginLeft: "50px",
  },
  calendarView: {
    height: "55px",
    alignItems: "center",
    marginBottom: "55px",
    marginLeft: "20px",
  },
  // New fc
  // split button
  splitButtonWrapper: {
    border: "1px solid #FF9212",
    borderRadius: "5px",
    color: "#FF9212",
    fontSize: "16px",
    fontWeight: "600",
    marginRight: "0px",
    background: "#FFFFFF 0% 0% no-repeat padding-box",
    "&:hover": {
      color: "#F68808",
      backgroundColor: "#FFFFFF",
      border: "1px solid #FF9212",
    },
  },
  splitButtonPaper: {
    width: "230px",
    height: "40px",
  },
  newFcdropdown: {
    marginRight: "25px",
  },
});

export default useStyles;
