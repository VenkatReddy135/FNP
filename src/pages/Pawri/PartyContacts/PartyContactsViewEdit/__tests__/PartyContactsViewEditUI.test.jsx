import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import renderer from "react-test-renderer";
import { ThemeProvider } from "@material-ui/core/styles";
import theme from "../../../../../assets/theme/theme";
import PartyContactsViewEditUI from "../PartyContactsViewEditUI";
const store = createStore(() => []);

const initialResponseObj = {
  toName: "Testing",
  attentionName: "Test",
  doorbell: "12",
  address1: "South gate, Pune",
  address2: "South gate, Pune",
  cityId: "jamshedpur",
  stateId: "IN-JH",
  pincode: "831002",
  countryId: "IND",
  contactPurposeId: "T_00092",
  contactTypeId: "T_00001",
  email: "",
  phone: "",
  fromDate: "2021-06-26T05:10:10",
  throughDate: "2021-06-27T05:10:10",
  countryCode: "+91",
};

describe("render without crashing the component", () => {
  it("renders", () => {
    render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <Router>
            <PartyContactsViewEditUI
              handleCityStateCountryChange={() => null}
              handleChange={() => null}
              updateContact={() => null}
              cancelHandler={() => null}
              responseData={initialResponseObj}
              isEditable
              isPincodeInvalid
              contactPurposes={[]}
              switchToEditHandler={() => null}
              countryList={[]}
              stateList={[]}
              citiesList={[]}
              countryCodes={[]}
              pinCodesInfo={[]}
              countryCodes={[]}
            />
          </Router>
        </Provider>
      </ThemeProvider>,
    );
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <Router>
              <PartyContactsViewEditUI
                handleCityStateCountryChange={() => null}
                handleChange={() => null}
                updateContact={() => null}
                cancelHandler={() => null}
                responseData={initialResponseObj}
                isEditable
                isPincodeInvalid
                contactPurposes={[]}
                switchToEditHandler={() => null}
                countryList={[]}
                stateList={[]}
                citiesList={[]}
                countryCodes={[]}
                pinCodesInfo={[]}
                countryCodes={[]}
              />
            </Router>
          </Provider>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
