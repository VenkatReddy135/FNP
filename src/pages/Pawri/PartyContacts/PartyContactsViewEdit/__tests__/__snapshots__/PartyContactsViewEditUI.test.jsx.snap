// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing the component matches snapshot 1`] = `
Array [
  <div
    className="MuiGrid-root makeStyles-titleGridStyle-108 MuiGrid-container MuiGrid-justify-xs-space-between"
  >
    <h5
      className="MuiTypography-root makeStyles-gridStyle-135 MuiTypography-h5"
    />
  </div>,
  <hr
    className="MuiDivider-root makeStyles-dividerStyle-149"
  />,
  <div
    className="MuiCardContent-root RaCardContentInner-root-181"
  >
    <div
      className="ra-field ra-field-undefined"
    >
      <form
        className="simple-form"
      >
        <div
          className="MuiCardContent-root RaCardContentInner-root-181"
        >
          <div
            className="ra-input ra-input-undefined"
          >
            <div
              className="MuiGrid-root RaFormInput-input-182 MuiGrid-container MuiGrid-spacing-xs-7 MuiGrid-item"
              mb={1}
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root makeStyles-textInputField-147 MuiFormControl-marginDense"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled Mui-disabled Mui-disabled"
                    data-shrink={false}
                    htmlFor="contactType"
                    id="contactType-label"
                  >
                    <span>
                      contact_type
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
                    onClick={[Function]}
                  >
                    <input
                      aria-describedby="contactType-helper-text"
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
                      disabled={true}
                      id="contactType"
                      name="contactType"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                  </div>
                  <p
                    className="MuiFormHelperText-root MuiFormHelperText-contained Mui-disabled MuiFormHelperText-marginDense"
                    id="contactType-helper-text"
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  aria-labelledby="downshift-1-label"
                  aria-owns={null}
                  className="RaAutocompleteInput-container-188"
                  role="combobox"
                >
                  <div
                    aria-activedescendant={null}
                    aria-autocomplete="list"
                    aria-controls={null}
                    aria-labelledby="downshift-1-label"
                    className="MuiFormControl-root MuiTextField-root makeStyles-autoCompleteItem-112 MuiFormControl-marginDense"
                    onKeyDown={[Function]}
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
                      data-shrink={false}
                      htmlFor="contactPurposeId"
                      id="downshift-1-label"
                    >
                      <span>
                        contact_purpose
                      </span>
                    </label>
                    <div
                      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
                      onClick={[Function]}
                    >
                      <input
                        aria-describedby="contactPurposeId-helper-text"
                        aria-invalid={false}
                        autoComplete="off"
                        autoFocus={false}
                        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
                        disabled={false}
                        id="contactPurposeId"
                        name="contactPurposeId"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="text"
                        value=""
                      />
                    </div>
                    <p
                      className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense"
                      id="contactPurposeId-helper-text"
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ra-input ra-input-undefined"
          >
            <div
              className="MuiGrid-root RaFormInput-input-182 makeStyles-gridMarginStyle-152 MuiGrid-item"
            >
              <div
                className="MuiFormGroup-root"
              >
                <label
                  className="MuiFormControlLabel-root"
                >
                  <span
                    className="MuiSwitch-root"
                  >
                    <span
                      aria-disabled={false}
                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-197 MuiSwitch-switchBase MuiSwitch-colorPrimary"
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <input
                          checked={false}
                          className="PrivateSwitchBase-input-200 MuiSwitch-input"
                          id="primary"
                          name="primary"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        <span
                          className="MuiSwitch-thumb"
                        />
                      </span>
                    </span>
                    <span
                      className="MuiSwitch-track"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    <span>
                      primary
                    </span>
                  </span>
                </label>
                <p
                  className="MuiFormHelperText-root"
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </p>
              </div>
            </div>
          </div>
          <div
            className="ra-input ra-input-undefined"
          >
            <div
              className="MuiGrid-root makeStyles-plusButton-162 MuiGrid-container MuiGrid-spacing-xs-7 MuiGrid-item"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root makeStyles-dateField-117 MuiFormControl-marginDense"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-filled Mui-disabled Mui-disabled MuiFormLabel-filled"
                    data-shrink={true}
                    htmlFor="fromDate"
                    id="fromDate-label"
                  >
                    <span>
                      from_date
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-disabled Mui-disabled MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
                    onClick={[Function]}
                  >
                    <input
                      aria-describedby="fromDate-helper-text"
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
                      disabled={true}
                      id="fromDate"
                      name="fromDate"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="date"
                      value="2021-06-26"
                    />
                  </div>
                  <p
                    className="MuiFormHelperText-root MuiFormHelperText-contained Mui-disabled MuiFormHelperText-filled MuiFormHelperText-marginDense"
                    id="fromDate-helper-text"
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <div
                  className="MuiFormControl-root MuiTextField-root makeStyles-dateField-117 MuiFormControl-marginDense"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-filled MuiFormLabel-filled"
                    data-shrink={true}
                    htmlFor="throughDate"
                    id="throughDate-label"
                  >
                    <span>
                      through_date
                    </span>
                  </label>
                  <div
                    className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
                    onClick={[Function]}
                  >
                    <input
                      aria-describedby="throughDate-helper-text"
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
                      disabled={false}
                      id="throughDate"
                      name="throughDate"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="date"
                      value="2021-06-27"
                    />
                  </div>
                  <p
                    className="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
                    id="throughDate-helper-text"
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiToolbar-root MuiToolbar-regular RaToolbar-toolbar-201 RaToolbar-mobileToolbar-203 MuiToolbar-gutters"
          role="toolbar"
        >
          <button
            aria-label="cancel"
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined RaButton-button-206 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="RaButton-label-207"
              >
                cancel
              </span>
            </span>
          </button>
          <button
            aria-label="update"
            className="MuiButtonBase-root MuiButton-root MuiButton-contained RaSaveButton-button-212 MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              update
            </span>
          </button>
        </div>
        <div
          className="RaToolbar-spacer-205"
        />
      </form>
    </div>
  </div>,
]
`;
